lib/active_model_serializers/rspec/matchers/schema.rb

```diff
@@ -0,0 +1,31 @@
+require 'active_model_serializers/test/schema'
+
+module ActiveModelSerializers
+  module RSpecMatchers
+    module Schema
+      extend ActiveSupport::Concern
+
+      included do
+        RSpec::Matchers.define :have_valid_schema do |schema_path|
```

Comment:
see https://github.com/rails-api/active_model_serializers/pull/1947#issuecomment-255118660 for some more ideas

It may be helpful for the matcher to rely on methods that can be overridden

In my experience, request and response matchers are different and it's very useful to validate headers.